core/src/main/scala/kafka/server/FinalizedFeatureChangeListener.scala

```diff
@@ -143,13 +142,17 @@ class FinalizedFeatureChangeListener(zkClient: KafkaZkClient) extends Logging {
   private class ChangeNotificationProcessorThread(name: String) extends ShutdownableThread(name = name) {
     override def doWork(): Unit = {
       try {
-        ignoring(classOf[InterruptedException]) {
-          queue.take.updateLatestOrThrow()
-        }
+        queue.take.updateLatestOrThrow()
       } catch {
         case e: Exception => {
-          error("Failed to process feature ZK node change event. The broker will eventually exit.", e)
-          throw new FatalExitError(1)
+          // We ignore InterruptedException. While the queue is empty and this thread is blocking on
+          // taking an item from the queue, a concurrent call to FinalizedFeatureChangeListener.close()
+          // could interrupt the thread and cause an InterruptedException to be raised from queue.take().
+          // In such a case, it is safe to ignore the exception since the thread is being shutdown.
+          if (!e.isInstanceOf[InterruptedException]) {
```

Comment:
How about using match pattern?
```scala
        case _: InterruptedException =>
          // We ignore InterruptedException. While the queue is empty and this thread is blocking on
          // taking an item from the queue, a concurrent call to FinalizedFeatureChangeListener.close()
          // could interrupt the thread and cause an InterruptedException to be raised from queue.take().
          // In such a case, it is safe to ignore the exception since the thread is being shutdown.
        case e: Exception =>
          error("Failed to process feature ZK node change event. The broker will eventually exit.", e)
          throw new FatalExitError(1)
```

Reply:
Done.